lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -138,8 +138,27 @@ def get_memory_facts(self):
         if None not in (memstats.get('swaptotal'), memstats.get('swapfree')):
             memstats['swap:used'] = memstats['swaptotal'] - memstats['swapfree']
 
+        memstats['real:physical'] = 0
+        if os.path.isdir('/sys/devices/system/edac/mc/mc0'):
+            for file in glob.glob('/sys/devices/system/edac/mc/*/dimm*/size'):
+                dimm_size_mb = get_file_content(file)
+                if dimm_size_mb is not None:
+                    memstats['real:physical'] += int(dimm_size_mb)
+        else:
+            dmi_bin = self.module.get_bin_path('dmidecode')
```

Comment:
since we already have a get_dmi_facts, i would leverage the work there to add to memory, if that data is missing after passing here

Reply:
I don't think thats going to work, the dmi facts are usually pulled from from /sys not dmidecode on modern systems, but the memory stuff needs to be pulled from dmidecode more often than the dmi facts, right?

I could pull all that dmidecode stuff out into another function if that would be better, but the code duplication is essentially 2 or 3 lines :)